python/raydp/tf/estimator.py

```diff
@@ -15,25 +15,26 @@
 # limitations under the License.
 #
 
-import json
-import os
 from typing import Any, List, NoReturn, Optional, Union
 
 import tensorflow as tf
 import tensorflow.keras as keras
-from ray.util.data import MLDataset
-from ray.util.sgd.tf import TFTrainer
-from ray.util.sgd.tf.tf_dataset import TFMLDataset
 from tensorflow import DType, TensorShape
-
+from tensorflow.keras.callbacks import Callback
+from ray import train
+from ray.train import Trainer
+from ray.train.tensorflow import prepare_dataset_shard
+from ray.data.dataset import Dataset
+from ray.data.impl.arrow_block import ArrowRow
 from raydp.estimator import EstimatorInterface
-from raydp.spark import RayMLDataset
 from raydp.spark.interfaces import SparkEstimatorInterface, DF, OPTIONAL_DF
-
+from raydp import stop_spark
+from raydp.spark import spark_dataframe_to_ray_dataset
 
 class TFEstimator(EstimatorInterface, SparkEstimatorInterface):
     def __init__(self,
                  num_workers: int = 1,
+                 parallelism: int = None,
```

Comment:
Perhaps we don't need to expose the parallelism parameter which seems to expose too much details. The default Spark parallelism works fine in our tests. Otherwise a user can always repartition the Spark DF to get the expected parallelism. Can we remove this parameter?

Reply:
Done.